templates/_schemas/details/default.twig

```diff
@@ -2,14 +2,29 @@
 
 
 
+{% set graph = create('Spatie\\SchemaOrg\\Graph') %}
+{% set seomateMetaData = craft.seomate.getMeta({
+  config: { includeSitenameInTitle: false },
+}|merge(seomate ?? {}, true)) %}
 {% set entryImages = entryImages ?? craft.assets.relatedTo(entry).kind('image').all() %}
 
 {% set defaultSchemaProps = {
-  title: entry.title,
+  title: seomateMetaData.meta.title ?? entry.title,
+  description: entry.description ?? seomateMetaData.meta.description,
   url: entry.url,
-  dateCreated: entry.postDate,
-  dateUpdated: entry.dateUpdated,
-  images: entryImages|filter|values,
+  mainEntityOfPage: mainEntityOfPage ?? null,
+  dateUpdated: entry.dateUpdated ?? null,
+  datePublished: entry.postDate ?? null,
+  dateCreated: entry.postDate ?? null,
+  image: entry.seoImage.one ?? entry.image.one ?? entry.imageRestricted.one ?? null,
```

Comment:
We are assuming the presence of custom project-specific fields in the boilerplate. Remove.

Reply:
Removed.

Reply:
Can we not get the SEO image info / URL from the SEOmate data and use that?

Reply:
Used from _Seomate_ data and kept entry related images as fallback.

Reply:
The operator then should be `?:` instead of `??`.

Reply:
Done.

But query:

Using `??` would mean if the value is defined and not `null` then return it otherwise return the else part and `?:` would mean if the value exists return it otherwise return the else part. What difference it's creating here to use `?:` in place of `??`?

Reply:
The null-coalescing operator `??` is useful when we’re trying to access a variable that might not exist. If we’re sure a variable will be defined, then it is better to use `?:` because the first operator, as its name says, only fallsback if the value is `null`. However, in a lot of cases the value may not be `null` but still non-useful. Eg: `false`, `''`, `[]`, `{}`. These are essentially empty or false-y values. We’re interested in non-empty content.

But you should verify if we can always rely on the `entrySeomateMeta.meta.image` property always being defined.

Reply:
Understood. Thanks for the explanation.

`entrySeomateMeta.meta.image` property will always be defined as we have set it in the plugin config for [`metaPropertyTypes`](https://github.com/miranj/craft-boilerplate/blob/01d4569ff925f46bfe9ae40ef946229beac79c96/config/seomate.php#L26).